src/main/scala/diplomaticobjectmodel/OMLogicalTree.scala

```diff
@@ -0,0 +1,72 @@
+// See LICENSE.SiFive for license details.
+
+package freechips.rocketchip.tile
+
+import freechips.rocketchip.diplomacy.ResourceBindingsMap
+import freechips.rocketchip.diplomaticobjectmodel.model.OMComponent
+import scala.collection.mutable.ArrayBuffer
+
+
+trait LogicalTree {
+  def getOMComponents(resourceBindingsMap: ResourceBindingsMap, children: Seq[OMComponent]): Seq[OMComponent]
+}
+
+/** LogicalTreeEdges hold LogicalTree Nodes which will be used to construct the logical tree.
+  * The LogicalTreeEdge is used to construct a parent child relationship between different modules.
+  *
+  * @param parent The parent LogicalTree node
+  * @param child The child LogicalTree node
+  */
+case class LogicalTreeEdge(
+  parent: LogicalTree,
+  child: LogicalTree
+)
+
+object LogicalModuleTree {
```

Comment:
What is Module-specific about this? Are logical tree different from the Module hierarchy tree, or not?

Reply:
LogicalTree's are elements of the LogicalModuleTree.

I am open to any names that people feel are appropriate.

Reply:
I would say `LogicalTreeNodes` are part of a `LogicalTree`

Reply:
Changed to LogicalTreeNode